tools/config_generator.py

```diff
@@ -79,11 +95,123 @@ def update_param(config_dict, arg_dict):
+                epilog=('Specify certain config (optional): ' +
+                        '[--<section>.<key>=<value>]'))
+    mode = parser.add_mutually_exclusive_group()
+    parser.add_argument('output_directory', metavar='output_directory',
+                        type=str, help='Directory to put configs')
+    mode.add_argument('--dataset', '-d', metavar='dataset', type=str,
+                      help='Dataset to preprocess')
+    mode.add_argument('--stats', '-s',
+                      metavar=('num_nodes', 'num_relations', 'num_train',
+                               'num_valid', 'num_test'),
+                      nargs=5, help='Dataset statistics\n' +
+                      'Enter in order of num_nodes, num_relations, num_train' +
+                      ' num_valid, num_test')
+    parser.add_argument('--device', '-dev', metavar='generate_config',
+                        choices=["GPU", "CPU", "multi-GPU"],
+                        nargs='?', default='GPU',
+                        help=('Generates configs for a single-GPU/multi-CPU' +
+                              '/multi-GPU training configuration file by ' +
+                              'default. \nValid options (default to GPU): ' +
+                              '[GPU, CPU, multi-GPU]'))
+
+    config_dict, valid_dict = read_template(DEFAULT_CONFIG_FILE)
+
+    for key in list(config_dict.keys())[1:]:
+        if valid_dict.get(key) is not None:
+            parser.add_argument(str("--" + key), metavar=key, type=str,
+                                choices=valid_dict.get(key),
+                                default=config_dict.get(key),
+                                help=argparse.SUPPRESS)
+        else:
+            parser.add_argument(str("--" + key), metavar=key, type=str,
+                                default=config_dict.get(key),
+                                help=argparse.SUPPRESS)
+
+    return parser, config_dict
+
+
+def parse_args(args):
+    arg_dict = vars(args)
+    set_up_files(args.output_directory)
+
+    arg_dict.update({"general.device": arg_dict.get("device")})
+    if arg_dict.get("device") == "multi-GPU":
+        arg_dict.update({"device": "multi_GPU"})
+    else:
+        arg_dict.update({"device": arg_dict.get("device")})
+
+    if arg_dict.get("general.random_seed") == "#":
+        arg_dict.pop("general.random_seed")
+
+    if arg_dict.get("dataset") is not None:
+        arg_dict.update({"dataset": arg_dict.get("dataset")})
+        arg_dict = update_dataset_stats(arg_dict.get("dataset"), arg_dict)
+    elif arg_dict.get("stats") is not None:
+        arg_dict = update_stats(arg_dict.get("stats"), arg_dict)
+    else:
+        raise RuntimeError("Must specify either dataset or dataset stats.")
+
+    return arg_dict
+
+
 if __name__ == "__main__":
-    print("This is a config generator.")
+    parser, config_dict = set_args()
+    args = parser.parse_args()
+    config_dict = parse_args(args)
+
+    dir = args.output_directory
```

Comment:
This sets the path options assuming that the data is located in the same location as the configuration file. This might not always be the case. We should add another option "data_directory" which points to the location of the files. If this isn't specified we can default to the "output_directory".